### PR TITLE
Update ternary_search.py

### PR DESCRIPTION
**Pull request by zyrch**
### **Describe your change:**

Reduces the overall complexity of the algorithm for discrete searches (like in a list)

* [ ] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [ ] This pull request is all my own work -- I have not plagiarized.
* [ ] I know that pull requests will not be merged if they fail the automated tests.
* [ ] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [ ] All new Python files are placed inside an existing directory.
* [ ] All filenames are in all lowercase characters with no spaces or dashes.
* [ ] All functions and variable names follow Python naming conventions.
* [ ] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [ ] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: searches/ternary_search.py [hunk 1]
```diff
@@ -31,8 +31,8 @@ def ite_ternary_search(A: List[int], target: int):
             if right - left < precision:
                 return lin_search(left, right, A, target)
 
-            oneThird = (left + right) / 3 + 1
-            twoThird = 2 * (left + right) / 3 + 1
+            oneThird = (left + right) / 2 - 1
+            twoThird =  (left + right) / 2 + 1
```
Comment (peteryao7) [inline searches/ternary_search.py:35]: Ternary search is a divide and conquer algorithm that's supposed to split an array into three equal portions. Here, you're splitting it into two halves and a subarray of either 0 or 1 in the middle.

Using floating point division `/` over integer division `//` is causing an entirely different bug, but the index calculations themselves are fine.

Another way to calculate `oneThird` and `twoThird` that prevents overflow problems in other languages is by doing it this way:
`oneThird = left + (right - left) // 3`
`twoThird = right - (right - left) // 3`

File: searches/ternary_search.py [hunk 2]
```diff
@@ -58,8 +58,8 @@ def rec_ternary_search(left: int, right: int, A: List[int], target: int):
         if right - left < precision:
             return lin_search(left, right, A, target)
 
-        oneThird = (left + right) / 3 + 1
-        twoThird = 2 * (left + right) / 3 + 1
+        oneThird = (left + right) / 2 - 1
+        twoThird = (left + right) / 2 + 1
```
Comment (peteryao7) [inline searches/ternary_search.py:62]: Same as the iterative change above.

**Review by peteryao7 — CHANGES_REQUESTED** [review]
The changes don't make the functions ternary search anymore.

File: searches/ternary_search.py (hunk 1 above)
Comment (zyrch) [reply searches/ternary_search.py:35]: By dividing the search space into three in ternary search we are able to find the minimum or maximum of convex functions, (if it is not convex then binary search is directly applicable). Now comparing three values in the search space allows us to eliminate one portion at a time. Now to increase the efficiency of the program we can decrease the distance two intervals and still get the same result, this method is also known as dichotomous search.

Comment (peteryao7) [reply searches/ternary_search.py]: Binary and ternary search are examples of dichotomous search. What ternary search lets you do over binary search is omit two sections, or 2/3, of the search space on each iteration (as opposed to 1/2 in binary search), which is correctly done at the end of the loop:

```
elif target < A[oneThird]:
    right = oneThird - 1
elif A[twoThird] < target:
    left = twoThird + 1
else:
    left = oneThird + 1
    right = twoThird - 1
```

The goal of the function is simply to find a specified value, so binary search will suffice, but by changing how the boundaries are calculated and changing the time complexity to O(log2(n)), you implemented binary search, not ternary search.

Comment (zyrch) [reply searches/ternary_search.py]: You are correct I just realized we are searching in a sorted array, and my method is a better only in the case when ternary search is used to min or max of convex functions or data sets.

**Pull request closed without merging**